boa/src/builtins/string/mod.rs

```diff
@@ -479,39 +482,88 @@ impl String {
+                                (Some('\''), _) => {
+                                    // $'
+                                    let end_of_match = mat.end();
+                                    result.push_str(&primitive_val[end_of_match..]);
+                                }
+                                (Some(second), Some(third))
+                                    if second_is_digit && third_is_digit =>
+                                {
+                                    // $nn
+                                    let tens = second.to_digit(10).unwrap() as usize;
+                                    let units = third.to_digit(10).unwrap() as usize;
+                                    let nn = 10 * tens + units;
+                                    if nn == 0 || nn > m {
+                                        [Some(first), Some(second), chars.next()]
+                                            .iter()
+                                            .flatten()
+                                            .for_each(|ch: &char| result.push(*ch));
+                                    } else {
+                                        let group = match caps.get(nn) {
+                                            Some(text) => text.as_str(),
+                                            None => "",
+                                        };
+                                        result.push_str(group);
+                                        chars.next(); // consume third
+                                    }
+                                }
+                                (Some(second), _) if second_is_digit => {
+                                    // $n
+                                    let n = second.to_digit(10).unwrap() as usize;
+                                    if n == 0 || n > m {
+                                        result.push(first);
+                                        result.push(second);
+                                    } else {
+                                        let group = match caps.get(n) {
+                                            Some(text) => text.as_str(),
+                                            None => "",
+                                        };
+                                        result.push_str(group);
+                                    }
+                                }
+                                (Some('<'), _) => {
+                                    // $<
+                                    todo!("named capture groups")
+                                }
+                                _ => {
+                                    // $?, ? is none of the above
+                                    // we can consume second because it isn't $
+                                    result.push(first);
+                                    second.iter().for_each(|ch: &char| result.push(*ch));
+                                }
```

Comment:
```suggestion
                                (_, _) => {
                                    // $?, ? is none of the above
                                    // we can consume second because it isn't $
                                    result.push(first);
                                    if let Some(second) = second {
                                    	result.push(second);
                                    }
                                }
```
This is more readable, If second is garanteed to be a `char` (not `None`) we can do:
```rust
                                (Some(second), _) => {
                                    // $?, ? is none of the above
                                    // we can consume second because it isn't $
                                    result.push(first);
                                  	result.push(second);
                                }
```